src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -1241,11 +1273,27 @@ private void ApplySignalWorkflow(SignalWorkflow signal)
                     return;
                 }
 
-                await inbound.Value.HandleSignalAsync(new(
-                    Signal: signal.SignalName,
-                    Definition: signalDefn,
-                    Args: args,
-                    Headers: signal.Headers)).ConfigureAwait(true);
+                // Handle signal
+                LinkedListNode<string>? warnableExecution = null;
+                if (signalDefn.UnfinishedPolicy == HandlerUnfinishedPolicy.WarnAndAbandon)
+                {
+                    warnableExecution = warnableInProgressSignals.AddLast(signal.SignalName);
+                }
+                try
+                {
+                    await inbound.Value.HandleSignalAsync(new(
+                        Signal: signal.SignalName,
+                        Definition: signalDefn,
+                        Args: args,
+                        Headers: signal.Headers)).ConfigureAwait(true);
+                }
+                finally
+                {
+                    if (warnableExecution is { } toRemove)
+                    {
+                        warnableInProgressSignals.Remove(toRemove);
```

Comment:
Note that this is just going to remove the first entry that has the same name (since signal name is not unique among handler executions). So using `AddLast` above is arguably misleading, since the order loses meaning via this call. A `MultiSet` (i.e. `collections.Counter` in Python might be a good choice for in-progress signals here. In my PR I gave them a unique ID but perhaps that was unnecessary.

Reply:
> Note that this is just going to remove the first entry that has the same name (since signal name is not unique among handler executions)

Not true, note that `toRemove` is a `LinkedListNode`. It has nothing to do with the contents of the node. This collection was specifically chosen for the cheap node removal by node that .NET offers.

Reply:
Ohhh, right. Yep I was thinking `warnableExecution` was the string signal name.